src/browser/ui/ReactDOMComponent.js

```diff
@@ -232,6 +234,24 @@ function processChildContext(context, tagName) {
   return context;
 }
 
+var onlyNestTextTags = {
+  'option': true
+};
+
+function getChildrenTextContent(tag, children) {
+  var childrenContent = '';
+  ReactChildren.forEach(children, function (c) {
+    if ("production" !== process.env.NODE_ENV) {
+      ("production" !== process.env.NODE_ENV ? warning(
+        !ReactElement.isValidElement(c),
+        tag + ' can not have nested tags'
+      ) : null);
+    }
+    childrenContent += c;
```

Comment:
Sorry, forgot that React.Children.forEach used traverseAllChildren. I believe `c` shows up as `null` in those cases though, so you still need to make a change here? Ideally we can add a unit test for this.